Games/Minesweeper/Field.cpp

```diff
@@ -156,7 +156,13 @@ Field::Field(GUI::Label& flag_label, GUI::Label& time_label, GUI::Button& face_b
         int mine_count = config->read_num_entry("Game", "MineCount", 10);
         int rows = config->read_num_entry("Game", "Rows", 9);
         int columns = config->read_num_entry("Game", "Columns", 9);
-        set_field_size(rows, columns, mine_count);
+
+        // Do a quick sanity check to make sure the user hasn't tried anything crazy
+        if (mine_count > rows * columns)
+            set_field_size(9, 9, 10);
```

Comment:
@awesomekling This effectively resets the game to `Beginner` mode. It'd probably be best to not have these hardcoded, but I feel that's probably a refactor for another PR.

Reply:
This does not seem sufficient. #1261 mentions various other cases, like "zero mines", "zero sized field", etc.
Also I'm not sure what you would achieve with a `Difficulty` struct?

Reply:
> Also I'm not sure what you would achieve with a Difficulty struct?

Mostly avoiding nasty hard coded constants, thought YMMV as to whether or not you're okay with them. We could have something like `set_field_size(m_current_difficulty)` or something like that. Just my 2c though.